PluginsAndFeatures/azure-toolkit-for-eclipse/com.microsoft.azuretools.appservice/src/com/microsoft/azure/toolkit/eclipse/function/wizard/ProjectArtifactPage.java

```diff
@@ -0,0 +1,159 @@
+import org.apache.commons.lang3.StringUtils;
+import org.eclipse.swt.SWT;
+import org.eclipse.swt.layout.GridData;
+import org.eclipse.swt.layout.GridLayout;
+import org.eclipse.swt.widgets.Composite;
+import org.eclipse.swt.widgets.Label;
+import org.eclipse.swt.widgets.Text;
+
+import javax.annotation.Nonnull;
+import java.util.Arrays;
+import java.util.List;
+import java.util.function.Predicate;
+import java.util.stream.Stream;
+
+public class ProjectArtifactPage extends AzWizardPageWrapper implements AzureForm<FunctionArtifactModel> {
+    private AzureTextInput txtGroupId;
+    private Composite container;
+    private AzureTextInput txtArtifact;
+    private AzureTextInput txtVersion;
+    private AzureTextInput txtPackageName;
+    private static final String PACKAGE_NAME_REGEX = "[a-zA-Z]([\\.a-zA-Z0-9_])*";
+    private static final String GROUP_ARTIFACT_ID_REGEX = "[0-9a-zA-Z]([\\.a-zA-Z0-9\\-_])*";
+    private static final String VERSION_REGEX = "[0-9]([\\.a-zA-Z0-9\\-_])*";
+
+    public ProjectArtifactPage() {
+        super("Artifact Page");
+        setTitle("New Azure Function Project");
+        setDescription("Provide the group id and artifact id");
+        setControl(txtGroupId);
+    }
+
+    public void setDefaultValueFirstTime() {
+        FunctionProjectPage previousPage = (FunctionProjectPage) getWizard().getPreviousPage(this);
+
+        if (Stream.of(txtArtifact, txtVersion, txtPackageName, txtGroupId).map(AzureTextInput::getValue).noneMatch(StringUtils::isNotBlank)) {
+            FunctionArtifactModel defaultModel = FunctionArtifactModel.getDefaultFunctionProjectConfig();
+            defaultModel.setArtifactId(previousPage.getProjectName());
+            setValue(defaultModel);
+        }
+        addValidator("Group id", txtGroupId, ProjectArtifactPage::isValidGroupIdArtifactId);
+        addValidator("Artifact id", txtArtifact, ProjectArtifactPage::isValidGroupIdArtifactId);
+        addValidator("Artifact id", txtVersion, ProjectArtifactPage::isValidVersion);
+        addValidator("Package name", txtPackageName, ProjectArtifactPage::isValidJavaPackageName);
+    }
+
+    private static void addValidator(String propertyName, AzureTextInput input, Predicate<String> validator) {
+        input.setRequired(true);
+        input.addValidator(() -> validateProperties(propertyName, input, validator));
+    }
+
+    private static AzureValidationInfo validateProperties(String propertyName, AzureTextInput input, Predicate<String> validator) {
+        String text = input.getValue();
+
+        if (text.isEmpty()) {
+            return AzureValidationInfo.builder().input(input).type(AzureValidationInfo.Type.ERROR).message(propertyName + " is required.").build();
+        }
+
+        if (!validator.test(text)) {
+            return AzureValidationInfo.builder().input(input).type(AzureValidationInfo.Type.ERROR).message(String.format("Invalid %s: %s", propertyName, text)).build();
```

Comment:
It could be better if we could tell user the regex to follow

Reply:
should be readable text.